plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.ts

```diff
@@ -18,18 +18,26 @@ import { GroupEntity } from '@backstage/catalog-model';
 import * as winston from 'winston';
 import { Config } from '@backstage/config';
 import { OktaEntityProvider } from './OktaEntityProvider';
+import {GroupNamingStrategy, groupNamingStrategyFactory} from './groupNamingStrategyFactory';
 
 /**
  * Provides entities from Okta Group service.
  */
 export class OktaGroupEntityProvider extends OktaEntityProvider {
-  static fromConfig(config: Config, options: { logger: winston.Logger }) {
+  private readonly namingStrategy: GroupNamingStrategy;
+
+  static fromConfig(config: Config, options: { logger: winston.Logger, namingStrategy?: string }) {
     const orgUrl = config.getString('orgUrl');
     const token = config.getString('token');
 
     return new OktaGroupEntityProvider({ orgUrl, token }, options);
   }
 
+  constructor(accountConfig: any, options: {  logger: winston.Logger, namingStrategy?: string }) {
+    super(accountConfig, options);
+    this.namingStrategy = groupNamingStrategyFactory(options.namingStrategy || 'id')
```

Comment:
default this (and the corresponding `userNamingStrategyFactory` on the method itself so this guy can safely pass in undefined  in case config is errorenous and doesn't need to be aware of the internals of the factory.

Reply:
done I think.